### PR TITLE
LPS-146861 Allow to sort entries by a field created after the object is published

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related Issue: https://issues.liferay.com/browse/LPS-146861;

Actually, this bug happens if the Object overall will not save, after the new Field is created. Then _entityModelMap_ is not updated with the new Fields.

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by gabrielwas** [comment]
Just started reviewing :)

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 1]
```diff
@@ -126,7 +129,7 @@ public ObjectEntry getByExternalReferenceCode(String externalReferenceCode)
 
 	@Override
 	public EntityModel getEntityModel(MultivaluedMap multivaluedMap) {
-		if (_entityModel == null) {
+		if ((_entityModel == null) || !_isEntityModelUpdated()) {
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:132]: @p-albuquerque As a matter of simplicity, we should remove this verification and always build the EntityModel for now...

Verifying if there's a change every time can be costly, you know?

**Review by gabrielwas — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 2]
```diff
@@ -125,11 +125,9 @@ public ObjectEntry getByExternalReferenceCode(String externalReferenceCode)
...
 	@Override
 	public EntityModel getEntityModel(MultivaluedMap multivaluedMap) {
-		if (_entityModel == null) {
-			_entityModel = new ObjectEntryEntityModel(
-				_objectFieldLocalService.getObjectFields(
-					_objectDefinition.getObjectDefinitionId()));
-		}
+		_entityModel = new ObjectEntryEntityModel(
+			_objectFieldLocalService.getObjectFields(
+				_objectDefinition.getObjectDefinitionId()));
```
Comment (gabrielwas) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:130]: @p-albuquerque can we return right away? Thx

Comment (p-albuquerque) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java]: Okay.

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
LGTM :call_me_hand::skin-tone-3:

**Pull request closed without merging**